packages/react-utilities/src/descendants/descendants.ts

```diff
@@ -0,0 +1,72 @@
+import * as React from 'react';
+import { useId, useFirstMount, useIsomorphicLayoutEffect } from '../hooks/index';
+
+export type Descendant = { id: string; forceUpdate: () => void };
+export type Descendants = Record<string, Descendant>;
+export type SetDescendant = (descendant: Descendant) => number;
+
+export interface DescendantsContextValue {
+  /**
+   * A record of descendants that have been registered by their id
+   */
+  descendants: Descendants;
+  /**
+   * add a descendant, if it hasn't been added before
+   * @returns index of descendant
+   */
+  setDescendant: SetDescendant;
+}
+
+const DescendantsContext = React.createContext<DescendantsContextValue | undefined>(undefined);
+
+/**
+ * hook to initialize descendants,
+ * this hook ensures that although new descendants can be added,
+ * no re-rendering will be triggered.
+ * The only way to trigger a re-rendering is by updating the parent itself,
+ * e.g: adding a new descendant, removing a descendant, re-ordering descendants, ...
```

Comment:
Hmmmm if I understand the purpose of this new API correctly the doc should be smth like this:

> Adding descendants with `setDescendant` will not trigger re-renders. A component that wants to update its list of descendats should force itself to rerender to get the latest order of descendants

Reply:
I'll update the doc, it's actually not even true anymore that concept of not re-rendering on adition